examples/arkode/F2003_serial/ark_roberts_dnsL_f2003.f90

```diff
@@ -0,0 +1,706 @@
+
+    integer(c_long), parameter :: neq = 3
+    integer(c_long), parameter :: nout = 12
+
+  contains
+
+    ! ----------------------------------------------------------------
+    ! fcnirob: The implicit RSH operator function
+    !
+    ! Return values:
+    !    0 = success,
+    !    1 = recoverable error,
+    !   -1 = non-recoverable error
+    ! ----------------------------------------------------------------
+    integer(c_int) function fcnirob(tres, sunvec_y, sunvec_f, user_data) &
+         result(ierr) bind(C,name='fcnirob')
+
+      !======= Inclusions ===========
+      use, intrinsic :: iso_c_binding
+      use fsundials_nvector_mod
+      use fnvector_serial_mod
+
+      !======= Declarations =========
+      implicit none
+
+      ! calling variables
+      real(c_double), value :: tres      ! current time
+      type(N_Vector)        :: sunvec_y  ! solution N_Vector
+      type(N_Vector)        :: sunvec_f  ! function N_Vector
+      type(c_ptr),    value :: user_data ! user-defined data
+
+      ! pointers to data in SUNDIALS vectors
+      real(c_double), pointer :: yval(:)
+      real(c_double), pointer :: fval(:)
+
+      !======= Internals ============
+
+      ! get data arrays from SUNDIALS vectors
+      yval  => FN_VGetArrayPointer(sunvec_y)
+      fval  => FN_VGetArrayPointer(sunvec_f)
+
+      ! fill residual vector
+      fval(1)  = -0.04d0*yval(1) + 1.0d4*yval(2)*yval(3)
+      fval(3)  = 3.0d7*yval(2)**2
+      fval(2)  = -fval(1) - fval(3)
+
+      ! return success
+      ierr = 0
+      return
+
+    end function fcnirob
+
+    ! ----------------------------------------------------------------
+    ! fcnerob: The explicit RHS operator function
+    !
+    ! Return values:
+    !    0 = success,
+    !    1 = recoverable error,
+    !   -1 = non-recoverable error
+    ! ----------------------------------------------------------------
+    integer(c_int) function fcnerob(tres, sunvec_y, sunvec_f, user_data) &
+         result(ierr) bind(C,name='fcnerob')
+
+      !======= Inclusions ===========
+      use, intrinsic :: iso_c_binding
+      use fsundials_nvector_mod
+      use fnvector_serial_mod
+
+      !======= Declarations =========
+      implicit none
+
+      ! calling variables
+      real(c_double), value :: tres      ! current time
+      type(N_Vector)        :: sunvec_y  ! solution N_Vector
+      type(N_Vector)        :: sunvec_f  ! function N_Vector
+      type(c_ptr),    value :: user_data ! user-defined data
+
+      ! pointers to data in SUNDIALS vectors
+      real(c_double), pointer :: yval(:)
+      real(c_double), pointer :: fval(:)
+
+      !======= Internals ============
+
+      ! get data arrays from SUNDIALS vectors
+      yval  => FN_VGetArrayPointer(sunvec_y)
+      fval  => FN_VGetArrayPointer(sunvec_f)
+
+      ! fill residual vector
+      fval(1)  = 0.d0
+      fval(2)  = 0.d0
+      fval(3)  = 0.d0
+
+      ! return success
+      ierr = 0
+      return
+
+    end function fcnerob
+
```

Comment:
This function does nothing, and is no longer required in our F2003 interfaces, so please delete.
```suggestion
```